WcaOnRails/lib/results_validators/advancement_conditions_validator.rb

```diff
@@ -29,50 +29,49 @@ def self.has_automated_fix?
       false
     end
 
-    def validate(competition_ids: [], model: Result, results: nil)
-      reset_state
-      # Get all results if not provided
-      results ||= model.sorted_for_competitions(competition_ids)
+    def competition_associations
+      {
+        rounds: [],
+      }
+    end
 
-      results_by_competition_id = results.group_by(&:competitionId)
+    def run_validation(validator_data)
+      ordered_round_type_ids = RoundType.order(:rank).all.map(&:id)
 
-      competitions_start_dates = Competition.where(id: results_by_competition_id.keys).select(:id, :start_date).to_h do |c|
-        [c.id, c.start_date]
-      end
+      validator_data.each do |competition_data|
+        competition = competition_data.competition
+        comp_start_date = competition.start_date
 
-      results_by_competition_id.each do |competition_id, results_for_comp|
-        competition = Competition.includes(:rounds).find(competition_id)
```

Comment:
`.includes(:rounds)` moved to the new `competition_associations` method that you can see on the right